include/solvers/navier_stokes_assemblers.h

```diff
@@ -294,6 +295,53 @@ class GDNavierStokesAssemblerCore : public NavierStokesAssemblerBase<dim>
   double                             gamma;
 };
 
+template <int dim>
+class GDNavierStokesAssemblerNonNewtonianCore
+  : public NavierStokesAssemblerBase<dim>
+{
+public:
+  GDNavierStokesAssemblerNonNewtonianCore(
+    std::shared_ptr<SimulationControl> simulation_control,
+    Parameters::PhysicalProperties     physical_properties,
+    const double                       gamma)
+    : simulation_control(simulation_control)
+    , physical_properties(physical_properties)
+    , gamma(gamma)
+  {
+    // if (physical_properties.non_newtonian_parameters.model ==
+    // Parameters::NonNewtonian::Model::Carreau)
+    //{
```

Comment:
Seems legit to leave it like that for the moment.

Reply:
Ok I'll leave it like this.